wgengine/userspace.go

```diff
@@ -1528,3 +1542,12 @@ func (e *userspaceEngine) InstallCaptureHook(cb capture.Callback) {
 	e.tundev.InstallCaptureHook(cb)
 	e.magicConn.InstallCaptureHook(cb)
 }
+
+func (e *userspaceEngine) reconfigureVPNIfNecessary() error {
+	if e.reconfigureVPN != nil {
+		if err := e.reconfigureVPN(); err != nil {
+			return err
+		}
+	}
+	return nil
```

Comment:
minor subjective style thing... Go generally tries to return early and outdent.  at least when it's too indented. This case isn't super bad, but shorter and less indented is:

```go
    if e.reconfigureVPN == nil {
         return nil
    }
    return e.reconfigureVPN()
```